webpack/assets/javascripts/react_app/routes/Audits/AuditsPage/AuditsPage.js

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable camelcase */
+import React, { Component } from 'react';
+import queryString from 'query-string';
+import PropTypes from 'prop-types';
+import { Button, Icon, Spinner } from 'patternfly-react';
+import './auditspage.scss';
+import { translate as __ } from '../../../common/I18n';
+
+import PageLayout from '../../common/PageLayout/PageLayout';
+import AuditsList from '../../../components/AuditsList';
+import Pagination from '../../../components/Pagination/Pagination';
+import DefaultEmptyState from '../../../components/common/EmptyState';
+import {
+  AUDITS_DOC_URL,
+  AUDITS_SEARCH_PROPS,
+  AUDITS_PER_PAGE_OPTIONS,
+} from './AuditsPageConstants';
+
+class AuditsPage extends Component {
+  fetchWithQuery = (query, historyAction) => {
+    const { history, fetchAudits } = this.props;
+    const { page = 1, per_page = 20, search = '' } = queryString.parse(query);
+    fetchAudits({
+      page,
+      perPage: per_page,
+      searchQuery: search,
+      history,
+      push: historyAction.push,
+      replace: historyAction.replace,
+    });
+  };
+
+  componentDidMount() {
+    const { location } = this.props;
+    const actions = {
+      push: false,
+      replace: true,
+    };
+    this.fetchWithQuery(location.search, actions);
+  }
+
+  componentDidUpdate(prevProps) {
+    const { location } = this.props;
```

Comment:
this trigger multiple API Calls, I don't understand why you need this "hack"

Reply:
Its temporary, but the problem is when the user presses the back button.
we need to handle this event efficiently (get the params from the url, then trigger an API call and change the pagination etc) I'm still figuring out the best cycle fo this.